File: javascript/src/televentas/detalleOrden.js
export class DetalleOrden {
  /**
   * @param {number} cantidad
   * @param {number} precioUnitario
   * @param {import('./producto.js').Producto} producto
   */
  constructor(cantidad, precioUnitario, producto) {
    this.cantidad = cantidad;
    this.precioUnitario = precioUnitario;
    this.producto = producto;
  }

  getCantidad() {
    return this.cantidad;
  }

  getPrecioUnitario() {
    return this.precioUnitario;
  }

  getProducto() {
    return this.producto;
  }

  calcularSubtotal() {
    return this.cantidad * this.precioUnitario;
  }
}

File: java/src/televentas/DetalleOrden.java
package televentas;

public class DetalleOrden {

    private final int cantidad;
    private final double precioUnitario;
    private final Producto producto;

    public DetalleOrden(int cantidad, double precioUnitario, Producto producto) {
        this.cantidad = cantidad;
        this.precioUnitario = precioUnitario;
        this.producto = producto;
    }

    public int getCantidad() {
        return cantidad;
    }

    public double getPrecioUnitario() {
        return precioUnitario;
    }

    public Producto getProducto() {
        return producto;
    }

    public double calcularSubtotal() {
        return cantidad * precioUnitario;
    }
}
